canonical-web-and-design/snapcraft.io
# Update jsdom to remove polyfills from tests

Fixes #1873 

- updates jsdom to 11.12.0 (latest minor version from ^11 that is used by jest)
- updates the tests to make sure they work with new jsdom:
  - adds testURL to config (to make sure jsdom localstorage implementation doesn't fail with security error)
  - updates tests that assert URL(hrefs, srcs) to make them work with testURL from config
  - removes localstorage mocks (now that jsdom implements localstorage)

### NOTE

I tried updating `jest` to bump minor version, but it started to fail. It looked like babel stopped transpiling test files correctly. Couldn't find if it's issue with our config, babel version or something else... We may keep that in mind when planning `jest` update in future, that it may not be a simple fix.

### QA
- ideally make a clean install (remove node_modules)
- `./run exec yarn install`
- `./run exec yarn test-js`
- all tests should work fine
